packages/sanity/src/core/studio/EnableErrorReporting.ts

```diff
@@ -0,0 +1,34 @@
+import {type SanityClient} from '@sanity/client'
+import {useEffect} from 'react'
+
+import {errorReporter} from '../error/errorReporter'
+import {useClient} from '../hooks'
+import {DEFAULT_STUDIO_CLIENT_OPTIONS} from '../studioClient'
+
+async function fetchTelemetryConsent(client: SanityClient) {
+  return client.request({uri: '/intake/telemetry-status'})
+}
+
+export function EnableErrorReporting() {
+  const client = useClient().withConfig(DEFAULT_STUDIO_CLIENT_OPTIONS)
+
+  useEffect(() => {
+    async function checkConsent() {
+      try {
+        const res = await fetchTelemetryConsent(client)
+
+        if (res?.status === 'granted' || res?.status === 'unset') {
```

Comment:
We should confirm if "unset" response is a sufficient response here.

Reply:
As far as I can tell from the code, `unset` means "not set, so don't allow", so this should be reduced to only `granted` I believe (ask me on Slack for details if you want a more thorough rationale)

Reply:
Sorted, happy to do so -- was thinking there was some complexity about this per region.

Reply:
Backend takes care of that - luckily